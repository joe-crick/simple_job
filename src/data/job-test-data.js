import faker from "faker";
import _ from "lodash";

export const jobData = num =>
  _.times(num, id => {
    return {
      id,
      companyName: faker.company.companyName(),
      companyLogo:
        "https://d3ey1qdi6d5u1h.cloudfront.net/images/5fa6176a-7887-4af3-9815-e59eff8c73f7/d5cfaddd-d305-45fb-b925-dc67782ee8c9.png",
      datePosted: faker.date.future().toLocaleDateString(),
      description: faker.lorem.sentence()
    };
  });
