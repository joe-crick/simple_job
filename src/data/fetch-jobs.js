import { jobData } from "./job-test-data";

export const fetchJobs = amount =>
  new Promise(resolve => setTimeout(() => resolve(jobData(amount)), 200));
