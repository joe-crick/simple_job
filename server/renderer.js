import React from "react";
import { renderToString } from "react-dom/server";
import Main from "../src/index";
import { jobData } from "../src/data/job-test-data";
import StaticRouter from "react-router-dom/StaticRouter";
import { Provider } from "react-redux";
import fs from "fs";
import path from "path";
import rootStore from "../src/stores/store";
import appConfig from "../src/app.config";
import defaultState from "../src/state/state";

const context = {};
const filePath = path.resolve(__dirname, "..", "build", "index.html");
const template = fs.readFileSync(filePath, "utf8");

export default (req, res) => {
  defaultState.jobsList = jobData(appConfig.defaultCount);
  const appHtml = renderToString(
    <Provider store={rootStore(defaultState)}>
      <StaticRouter location={req.url} context={context}>
        <Main />
      </StaticRouter>
    </Provider>
  );

  return context.url
    ? res.redirect(301, context.url)
    : res.send(
        template
          .replace("{{SSR}}", appHtml)
          .replace("{{json}}", JSON.stringify(defaultState.jobsList))
      );
};
