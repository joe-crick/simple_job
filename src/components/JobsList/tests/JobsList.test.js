import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import JobList, { isSentinel, jobCount } from "../JobsList";
import { jobData } from "../../../data/job-test-data";

const COUNT = 10;
const jobs = jobData(COUNT);

describe("JobsList", () => {
  it("should render", () => {
    const component = renderer.create(<JobList />);
    let header = component.toJSON();
    expect(header).toMatchSnapshot();
  });
  it("should display a list of jobsList when given an array of jobsList", () => {
    const expected = COUNT;
    const component = mount(<JobList jobs={jobs} />);
    const actual = component.find(".JobThumbnail").length;
    expect(actual).toEqual(expected);
  });
  it("should display nothing found message when not given jobsList", () => {
    const expected = 1;
    const component = mount(<JobList />);
    const actual = component.find(".no-jobs").length;
    expect(actual).toEqual(expected);
  });
  it("should mark the fifth to last row of data as a sentinel", () => {
    const expected = true;
    const component = mount(<JobList jobs={jobs} />);
    const firstJob = component.find(".JobThumbnail").at(jobs.length - 5);
    const actual = firstJob.hasClass("sentinel");
    expect(actual).toEqual(expected);
  });
});

describe("isSentinel", () => {
  it('should return "sentinel" if the first argument is 1, and the length is 6', () => {
    const expected = "sentinel";
    const actual = isSentinel(1, 6);
    expect(actual).toEqual(expected);
  });
  it('should return "" if the first argument is 1, and the length is 1', () => {
    const expected = "";
    const actual = isSentinel(1, 2);
    expect(actual).toEqual(expected);
  });
});

describe("jobCount", () => {
  it("should return the length of an array if one is provided", () => {
    const expected = 1;
    const test = [1];
    const actual = jobCount(test);
    expect(actual).toEqual(expected);
  });
  it("should return 0 if a falsy value is provided", () => {
    const expected = 0;
    const actual = jobCount();
    expect(actual).toEqual(expected);
  });
});
