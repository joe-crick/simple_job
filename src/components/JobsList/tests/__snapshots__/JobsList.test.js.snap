// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsList should render 1`] = `
<section
  className="Jobs"
>
  <div
    className="no-jobs"
  >
    No jobs found
  </div>
</section>
`;
